tree_hash.go

```diff
@@ -0,0 +1,267 @@
+package ssz
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+)
+
+const hashLengthBytes = 32
```

Comment:
Can you add a comment what this references in the spec? I think it's `BYTES_PER_CHUNK`.